src/entities/password-input/ui/password-input.tsx

```diff
@@ -0,0 +1,61 @@
+import { styled } from "@shared/ui/theme"
+import { IconLock1, IconEye, IconEyeOff } from "@shared/ui/icons"
+import { useTheme } from "@shared/hooks"
+import { Input } from "@shared/ui/atoms"
+import { useCallback, useState } from "react"
+
+type Props = {
+    password: string,
+    onPasswordChange: (password: string) => void
+}
+
+export const PasswordInput = ({ password, onPasswordChange }: Props) => {
+    const [isPasswordShowing, setPasswordShowing] = useState(false)
+    const theme = useTheme()
+
+    const onToggleSecureButtonClick = useCallback(() => {
+        setPasswordShowing(!isPasswordShowing)
```

Comment:
Пупупу
Такого делать не надо
Можешь попасться на неактульный стейт

setState функция может принимать колбэк, в котором всегда будет актуальный стейт

```suggestion
        setPasswordShowing(prev => !prev)
```